packages/datadog-instrumentations/src/playwright.js

```diff
@@ -181,18 +181,27 @@ function dispatcherHook (dispatcherExport) {
   return dispatcherExport
 }
 
+function getTestByTestId (dispatcher, testId) {
+  if (dispatcher._testById) {
+    return dispatcher._testById.get(testId)?.test
+  }
+  if (dispatcher._allTests) {
+    return dispatcher._allTests.find(({ id }) => id === testId)
```

Comment:
This looks like a linear-time search.
If this `dispatcher` entity has a lot of tests and we expect to use the function a lot, perhaps creating a simple index of our own is worth it (e.g. iterate over the tests once and place them in a map where ID is the key, then use the map for all subsequent lookups).